src/datasets/summarizations/trend-regression.summarization.ts

```diff
@@ -87,41 +100,44 @@ export function queryFactory(points: TimeSeriesPoint[]) {
       ['quickly decreasing', uQuicklyDecreasingLinearDynamic],
     ];
 
-    const summaries: Summary[] = [];
+    // TODO: Move denormalization information to normalization utils
+    const ymin = 0;
+    const ymax = Math.max(...points.map(({ y }) => y));
+    const xdiff = 800 / 500 / points.length;
+    const denormalizeGradient = gradient => (gradient * xdiff) * (ymax - ymin) - ymin;
+
     const normalizedPoints = normalizePoints(points.map(timeSeriesPointToNumPoint));
     const weekdayNormalizedPoints = normalizedPoints.filter((_, i) => isWeekday(points[i]));
     const weekendNormalizedPoints = normalizedPoints.filter((_, i) => isWeekend(points[i]));
 
     const overallLinearModel = createLinearModel(normalizedPoints);
-    const overallLinearTrendValidity = uSmallRegressionError(overallLinearModel.absoluteErrorStd);
+    const overallLinearTrendValidity = uSmallRegressionStd(overallLinearModel.errorStd);
 
     const weekdayLinearModel = createLinearModel(weekdayNormalizedPoints);
-    const weekdayLinearTrendValidity = uSmallRegressionError(weekdayLinearModel.absoluteErrorStd);
+    const weekdayLinearTrendValidity = uSmallRegressionStd(weekdayLinearModel.errorStd);
 
     const weekendLinearModel = createLinearModel(weekendNormalizedPoints);
-    const weekendLinearTrendValidity = uSmallRegressionError(weekendLinearModel.absoluteErrorStd);
-
-    const overallLinearTrendSummariesValidity = Math.max(
-      weekdayWeekendEqualValidity,
-      ...uLinearDynamics.map(([_, uLinearDynamic]) =>
-        Math.min(
-          overallLinearTrendValidity,
-          uLinearDynamic(overallLinearModel.gradientAngleRad),
-          weekdayLinearTrendValidity,
-          uLinearDynamic(weekdayLinearModel.gradientAngleRad),
-          weekendLinearTrendValidity,
-          uLinearDynamic(weekendLinearModel.gradientAngleRad),
-        )),
-    );
+    const weekendLinearTrendValidity = uSmallRegressionStd(weekendLinearModel.errorStd);
 
+    const overallLinearTrendSummariesValidity = weekdayWeekendEqualValidity;
+
+    const summaries: Summary[] = [];
     // Create summaries describing linear trend of overall points
     for (const [linearDynamic, uLinearDynamic] of uLinearDynamics) {
       const validity = Math.min(
         overallLinearTrendSummariesValidity,
         overallLinearTrendValidity,
         uLinearDynamic(overallLinearModel.gradientAngleRad),
       );
-      const text = `The <b>overall</b> trend is <b>${linearDynamic === 'constant' ? '' : 'linearly '}${linearDynamic}</b>.`;
+      const rate = denormalizeGradient(overallLinearModel.gradient);
+      const rateAbsolute = Math.abs(rate);
+
+      let text;
+      if (linearDynamic === 'constant') {
+        text = `The <b>overall</b> active users <b>remained similar</b>.`;
+      } else {
+        text = `The <b>overall</b> active users was <b>linearly ${linearDynamic}</b> by <b>${formatY(rateAbsolute)}</b> users per day.`;

```

Comment:
nit: these new summaries are not tested

Reply:
Nothing new in this PR. What I did in this PR is improving the algorithm that create the summaries to make it more robust to some real-world tricky cases. There are just small changes in summary templates and all covered by old tests.